motionSetPointsApp/src/setPoint.cpp

```diff
@@ -57,72 +62,93 @@ LookupTable& getTable(const char *env_fname) {
 //   const char *fname     [in] File to load
 //   const char *env_fname [in] Key to identify file
 void loadFile(const char *fname, const char *env_fname) {
+	std::set<std::string, CaselessCompare> read_names; // for checking uniqueness
 	FILE *fptr = fopen(fname, "rt");
 	int rowCount = 0;
 	char buff[ROW_LEN];
 	if ( fptr==NULL ) {
-		errlogPrintf("motionSetPoints: Unable to open lookup file \"%s\"\n", fname);
+		errlogSevPrintf(errlogMajor, "motionSetPoints: Unable to open lookup file \"%s\"\n", fname);
 		return;
 	}
-	//printf("Creating table: %s[%s]\n", env_fname, fname);
 	LookupTable& table = getTable(env_fname);
     epicsGuard<epicsMutex> _lock(g_lock); // need to protect write access to map	
 	table.rows.clear();
 	
+    setNumCoords(env_fname, 0);
     int numCoords = 0;
 	while ( fgets(buff, ROW_LEN, fptr) ) {
 		if ( buff[0]!='#' ) {
 			LookupRow row;
 			int count = sscanf(buff, "%39s %lf %lf", row.name, &row.x, &row.y);
 			if ( count<2 ) {
-				errlogPrintf("motionSetPoints: Error parsing %s line %d\n", fname, table.rows.size()+1);
+				errlogSevPrintf(errlogMajor, "motionSetPoints: Error parsing %s line %d\n", fname, table.rows.size()+1);
+				fclose(fptr);
+				table.rows.clear();
 				return;
 			}
             else if ( numCoords==0 ) {
                 numCoords = count - 1;
             }
             else if ( numCoords!=count-1 ) {
-				errlogPrintf("motionSetPoints: Inconsistent column count %s line %d\n", fname, table.rows.size()+1);
+				errlogSevPrintf(errlogMajor, "motionSetPoints: Inconsistent column count in %s line %d\n", fname, table.rows.size()+1);
+				fclose(fptr);
+				table.rows.clear();
 				return;
             }
+			if (read_names.count(row.name) != 0)
+			{
+				errlogSevPrintf(errlogMajor, "motionSetPoints: duplicate name \"%s\" in %s line %d\n", row.name, fname, table.rows.size()+1);
+				fclose(fptr);
+				table.rows.clear();
+				return;
+			}
+			for(int i = 0; i < table.rows.size(); ++i)
+			{
+				if (row.x == table.rows[i].x && row.y == table.rows[i].y)
+				{
+					errlogSevPrintf(errlogMajor, "motionSetPoints: duplicate coordinates for name \"%s\" in %s line %d\n", row.name, fname, table.rows.size()+1);
+				    fclose(fptr);
+					table.rows.clear();
+					return;
+				}
+			}
 			table.rows.push_back(row);
-			//printf("Row: %s\n", buff);
+			read_names.insert(row.name);
 		}
 	}
+	fclose(fptr);
     
     setNumCoords(env_fname, numCoords);
 
 	if ( table.rows.size()==0 ) {
-		errlogPrintf("motionSetPoints: Lookup file %s contains no rows\n", fname);
-		return;
+		errlogSevPrintf(errlogMinor, "motionSetPoints: Lookup file %s contains no rows\n", fname);
```

Comment:
Should this be major severity? I'm struggling to think of a case where someone intentionally loads a file with no rows.